pandas/core/dtypes/concat.py

```diff
@@ -320,45 +327,3 @@ def _maybe_unwrap(x):
 
     dtype = CategoricalDtype(categories=categories, ordered=ordered)
     return Categorical._simple_new(new_codes, dtype=dtype)
-
-
-def _concatenate_2d(to_concat: Sequence[np.ndarray], axis: AxisInt) -> np.ndarray:
-    # coerce to 2d if needed & concatenate
-    if axis == 1:
-        to_concat = [np.atleast_2d(x) for x in to_concat]
-    return np.concatenate(to_concat, axis=axis)
-
-
-def _concat_datetime(to_concat: Sequence[ArrayLike], axis: AxisInt = 0) -> ArrayLike:
-    """
-    provide concatenation of an datetimelike array of arrays each of which is a
-    single M8[ns], datetime64[ns, tz] or m8[ns] dtype
-
-    Parameters
-    ----------
-    to_concat : sequence of arrays
-    axis : axis to provide concatenation
-
-    Returns
-    -------
-    a single array, preserving the combined dtypes
-    """
-    from pandas.core.construction import ensure_wrapped_if_datetimelike
-
-    to_concat = [ensure_wrapped_if_datetimelike(x) for x in to_concat]
```

Comment:
also, could you explain why `ensure_wrapped_if_datetimelike` and `atleast_2d` are no longer needed here?

in general, for anything non-trivial, I'd really appreciate a comment explaining why you're making changes, else reviews can really take hours

Reply:
the `not single_dtype` case is irrelevant bc in that case we find_common_dtype and cast to it.  The ensure_wrapped_if_datetimelike is unnecessary bc caller is responsible for ensuring we are wrapped where appropriate.  The axis is unnecessary bc the axis=1 cases all go through the lib.dtypes_all_equal path at the top of concat_compat

Reply:
> The axis is unnecessary bc the axis=1 cases all go through the lib.dtypes_all_equal path at the top of concat_compat

Not sure I follow, sorry - in `test_append_new_columns`, it reaches `concat_compat` with `axis=1` but it doesn't go through the `# fastpath!` path at the top of `concat_compat`

Reply:
test_append_new_columns doesnt have any datetime columns, so wouldn't go through _concat_datetime.  Should have been clearer thats what i was referring to.